pkg/operator/ceph/cluster/crash/crash.go

```diff
@@ -164,14 +164,18 @@ func (r *ReconcileNode) createOrUpdateCephCron(cephCluster cephv1.CephCluster, c
 
 	// minimum k8s version required for v1 cronJob is 'v1.21.0'. Apply v1 if k8s version is at least 'v1.21.0', else apply v1beta1 cronJob.
 	if useCronJobV1 {
-		// delete v1beta1 cronJob if it already exists
-		err := r.client.Delete(r.opManagerContext, &v1beta1.CronJob{ObjectMeta: objectMeta})
-		if err != nil && !apierrors.IsNotFound(err) {
-			return controllerutil.OperationResultNone, errors.Wrapf(err, "failed to delete CronJob v1Beta1 %q", prunerName)
+		// delete v1beta1 cronJob if v1Job is not found
```

Comment:
I think you can make it more clear here,

// delete v1beta1 cronJob on an update to v1 cronjob, 
only if v1Job is still not created